<?xml version="1.0"?>
<launch>
  <arg name="gdb_debug" default="false"/>
  <arg name="profile" default="false"/>

  <arg name="kalibr_yaml"/>
  <arg name="kalibr_camera" default="cam0"/>

  <group ns="msckf">
    <param name="kalibr_camera_name" value="$(arg kalibr_camera)"/>
    <rosparam command="load" file="$(arg kalibr_yaml)"/>

    <param name="feature_cov" value="8"/>

    <param name="imu_vars/w_var" value="1e-4"/>
    <param name="imu_vars/dbg_var" value="3.6733e-5"/>
    <param name="imu_vars/a_var" value="1e-2"/>
    <param name="imu_vars/dba_var" value="7e-2"/>

    <param name="imu_covars/p_init" value="1e-12"/>
    <param name="imu_covars/q_init" value="1e-5"/>
    <param name="imu_covars/v_init" value="1e-2"/>
    <param name="imu_covars/ba_init" value="1e-2"/>
    <param name="imu_covars/bg_init" value="1e-2"/>

    <param name="max_gn_cost_norm" value="7"/>
    <param name="translation_threshold" value="0.1"/>

    <param name="keyframe_transl_dist" value="0.5"/>
    <param name="keyframe_rot_dist" value="0.5"/>

    <param name="min_track_length" value="5"/>
    <param name="max_track_length" value="50"/>
    <param name="max_cam_states" value="30"/>

    <param name="ransac_threshold" value="0.00000002"/>

    <param name="n_grid_rows" value="10"/>
    <param name="n_grid_cols" value="10"/>

    <group unless="$(arg profile)">
      <node pkg="msckf_mono" name="msckf_mono_node" type="msckf_mono_node" unless="$(arg gdb_debug)" output="screen"/>
      <node pkg="msckf_mono" name="msckf_mono_node" type="msckf_mono_node" launch-prefix="xterm -e gdb --args" if="$(arg gdb_debug)"/>
    </group>
    <group if="$(arg profile)">
      <node pkg="msckf_mono" name="msckf_mono_node" type="msckf_mono_node" launch-prefix="perf record -g --call-graph dwarf --output=perf.out.node_name.data --" if="$(arg profile)"/>
    </group>

  </group>
</launch>
